webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/NodeFactQueryProcessor.scala

```diff
@@ -216,15 +242,20 @@ class NodeFactQueryProcessor(
   // For now, we have to check the criterion to know, each external service is a special case
   def ldapMatcher(lines: Seq[CriterionLine], q: Query): Seq[IOResult[NodeFactMatcher]] = {
     // for LDAP, we rebuild a false query with only these lines and no transformation
-    val ldapQuery = q.copy(transform = ResultTransformation.Identity, criteria = lines.toList)
-    Seq(for {
-      nodeIds <- ldapQueryProc.rawInternalQueryProcessor(ldapQuery)
-    } yield {
-      NodeFactMatcher(
-        s"[sub-ldap query ${ldapQuery.toString()}]",
-        (n: CoreNodeFact) => nodeIds.contains(n.id).succeed
-      )
-    })
+    if (lines.nonEmpty) {
```

Comment:
The second big problem is solved here: when there is no LDAP line, we just don't create any LDAP matcher at all, period.